### PR TITLE
Add support for generating HomeKit ADK documentation

### PR DESCRIPTION
**Pull request by aajain-com**
Fixes: https://github.com/apple/HomeKitADK/issues/38

### CODE REVIEW

File: Documentation/conf.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
```
Comment (Clovel) [inline Documentation/conf.py:1]: This means that python is now a project dependency. The container dependencies must be updated in the `Doxkerfile`s. 

That and all Sphnx related dependencies should be added as well.

Comment (aajain-com) [reply Documentation/conf.py:1]: All required dependencies are installed by `./Tools/generate_api_doc.sh `. Reason this is done in the script and not inside docker is to keep the docker build small. Any tools and linter dependencies are installed by the respective scripts themselves when run the first time.

Comment (aajain-com) [reply Documentation/conf.py:1]: Script that generates the documentation below install all the required dependencies.

Comment (Clovel) [reply Documentation/conf.py:1]: Ok, I didn’t catch that.

**Pull request merged**